server/api/course.js

```diff
@@ -18,8 +20,18 @@ router.use((req, res, next) => {
 
 router.get("/", async (req, res, next) => {
   try {
-    let courses = await Course.find(req.campusFilter);
-    return res.json(courses);
+    let courses;
+    if (req.query.prof) {
+      let prof = await Prof.find({ email: req.query.prof });
+      if (!prof) return res.status(404).json([]);
+      courses = await Course.find({
+        ...req.campusFilter,
+        "history.professor": prof
+      }).select("id name");
```

Comment:
Selection of fields will be handled by mongoose pre-hook, so this is not required here

Reply:
but we don't need to send the course history in this case.